File: app/src/main/java/com/example/engineeringthesis/SelectLanguageActivity.java
package com.example.engineeringthesis;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ListView;

import com.example.engineeringthesis.Adapter.SelectLanguageAdapter;

import java.nio.channels.InterruptedByTimeoutException;

public class SelectLanguageActivity extends AppCompatActivity {
        ListView lv;
        String languages[] = {"Angielski","Niemiecki"};
        Integer languagesFlag[]={R.drawable.ang_icon,R.drawable.niem_icon};
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_select_language);
        lv = findViewById(R.id.selectLanguageListView);
        SelectLanguageAdapter selectLanguageAdapter = new SelectLanguageAdapter(SelectLanguageActivity.this,languages,languagesFlag);
        lv.setAdapter(selectLanguageAdapter);
        lv.setOnItemClickListener(new AdapterView.OnItemClickListener() {
            @Override
            public void onItemClick(AdapterView<?> adapterView, View view, int i, long l)
            {
                Intent intent = new Intent(SelectLanguageActivity.this,CategoryActivity.class);
                startActivity(intent);
            }
        });
    }
}

File: app/src/main/java/com/example/engineeringthesis/SelectLanguageActivity.kt
package com.example.engineeringthesis

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.ViewModelProvider
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.example.engineeringthesis.adapter.SelectLanguageAdapter
import com.example.engineeringthesis.adapter.SelectLanguageAdapter.OnLanguageListener
import com.example.engineeringthesis.viewmodel.LanguageViewModel

class SelectLanguageActivity : AppCompatActivity(), OnLanguageListener {
    private var languageViewModel: LanguageViewModel? = null
    var languagesFlag = arrayOf(R.drawable.ang_icon, R.drawable.niem_icon)

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_select_language)
        languageViewModel = ViewModelProvider(this).get(LanguageViewModel::class.java)
        val recyclerView = findViewById<RecyclerView>(R.id.languageRecyclerView)
        recyclerView.layoutManager = LinearLayoutManager(this)
        recyclerView.setHasFixedSize(true)
        val languageAdapter = SelectLanguageAdapter(this)
        recyclerView.adapter = languageAdapter
        languageViewModel!!.allLanguages.observe(this, { languages -> languageAdapter.setLanguagesList(languages) })
    }

    override fun onLanguageClick(position: Int) {
        val intent = Intent(this, CategoryActivity::class.java)
        startActivity(intent)
    }
}